packages/babel-parser/src/tokenizer/context.js

```diff
@@ -140,3 +141,8 @@ tt.backQuote.updateContext = function () {
 tt.star.updateContext = function () {
   this.state.exprAllowed = false;
 };
+
+tt.braceHashL.updateContext = function () {
```

Comment:
Nit: can you add a comment that we don't do this for `braceBarL` because we only pop the context when we see `}` and not `|}`?